java/src/main/java/com/logicalclocks/hsfs/engine/HudiEngine.java

```diff
@@ -0,0 +1,293 @@
+    hudiArgs.put(Constants.HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY,
+                Constants.DEFAULT_HIVE_PARTITION_EXTRACTOR_CLASS_OPT_VAL);
+    // Hive
+    hudiArgs.put(Constants.HUDI_HIVE_SYNC_ENABLE, "true");
+    hudiArgs.put(Constants.HUDI_HIVE_SYNC_TABLE, tableName);
+    // Gets the JDBC Connector for the project's Hive metastore
+    StorageConnectorApi storageConnectorApi = new StorageConnectorApi();
+    StorageConnector storageConnector = storageConnectorApi.getByNameAndType(featureGroup.getFeatureStore(),
+                featureGroup.getFeatureStore().getName(), StorageConnectorType.JDBC);
+    String connStr = storageConnector.getConnectionString();
+    String pw = FileUtils.readFileToString(new File("material_passwd"));
+    String jdbcUrl = connStr + "sslTrustStore=t_certificate;trustStorePassword="
+                + pw + ";sslKeyStore=k_certificate;keyStorePassword=" + pw;
+    hudiArgs.put(Constants.HUDI_HIVE_SYNC_JDBC_URL, jdbcUrl);
+    hudiArgs.put(Constants.HUDI_HIVE_SYNC_DB, featureGroup.getFeatureStore().getName());
+
+    return hudiArgs;
+  }
+
+  //hudi time time travel sql query
+  public void registerTemporaryTable(SparkSession sparkSession, FeatureGroup featureGroup, String alias,
+                                     Long startTimestamp, Long  endTimestamp) {
+
+    sparkSession.conf().set("spark.sql.hive.convertMetastoreParquet", "false");
+    sparkSession.sparkContext().hadoopConfiguration().setClass("mapreduce.input.pathFilter.class",
+        org.apache.hudi.hadoop.HoodieROTablePathFilter.class, org.apache.hadoop.fs.PathFilter.class);
+
+    Map<String, String> hudiArgs = setupHudiReadOpts(featureGroup, startTimestamp, endTimestamp);
+    sparkSession.read()
+        .format(Constants.HUDI_SPARK_FORMAT)
+        .options(hudiArgs)
+        .load(featureGroup.getLocation()).registerTempTable(alias);
+  }
+
+  private Map<String, String> setupHudiReadOpts(FeatureGroup featureGroup, Long startTimestamp, Long endTimestamp) {
+    Map<String, String> hudiArgs = new HashMap<String, String>();
+    Integer numberOfpartitionCols = utils.getPartitionColumns(featureGroup).length();
+    String basePath = featureGroup.getLocation();
+
+    //Snapshot query
+    if (startTimestamp == null && endTimestamp == null) {
+      hudiArgs.put(Constants.HUDI_QUERY_TYPE_OPT_KEY, Constants.HUDI_QUERY_TYPE_SNAPSHOT_OPT_VAL);
+      basePath =  basePath + StringUtils.repeat("/*", numberOfpartitionCols + 1);
+    } else if (endTimestamp != null) {
+      hudiArgs.put(Constants.HUDI_QUERY_TYPE_OPT_KEY, Constants.HUDI_QUERY_TYPE_INCREMENTAL_OPT_VAL);
+      String hudiCommitEndTime = timeStampToHudiFormat(endTimestamp);
+      //point in time  query
+      if (startTimestamp == null) {
+        hudiArgs.put(Constants.HUDI_BEGIN_INSTANTTIME_OPT_KEY, "000");
+        hudiArgs.put(Constants.HUDI_END_INSTANTTIME_OPT_KEY, hudiCommitEndTime);
+      } else if (startTimestamp != null) {       //incremental  query
```

Comment:
```suggestion
      } else {       //incremental  query
```
`startTimestamp != null` is always true if we reach this point.